nano/node/json_handler.cpp

```diff
@@ -3679,7 +3679,7 @@ void nano::json_handler::republish ()
 							while (block_d != nullptr && hash != source)
 							{
 								hashes.push_back (previous);
-								source = node.ledger.block_source (transaction, *block_d);
+								source = block_d->source_field ().value_or (block_d->is_send () ? 0 : block_d->link_field ().value_or (0).as_block_hash ());
```

Comment:
A short `node.ledger.block_source ( .. )` call changed into this relatively long conditional statement, is this intentional?